mm2src/mm2_main/src/lp_swap/saved_swap.rs

```diff
@@ -239,7 +270,11 @@ mod native_impl {
         }
 
         async fn save_to_db(&self, ctx: &MmArc) -> SavedSwapResult<()> {
-            let path = my_swap_file_path(ctx, self.uuid());
+            #[cfg(feature = "new-db-arch")]
+            let address_dir = self.address_dir();
+            #[cfg(not(feature = "new-db-arch"))]
+            let address_dir = "no address directory for old DB architecture (has no effect)";
```

Comment:
:thinking: 

How?

Reply:
any method taking an address dir will eventually call `ctx.address_dir(addr)` down the call stack (but don't trust, verify :)).

`ctx.address_dir(addr)` will just return `ctx.dbdir()` for `not(new-db-arch)` targets regardless of `addr`. so any `address_dir` has no effect for `not(new-db-arch)` and is just a dummy variable that helps us not duplicate a lot of function definition and usage.